lib/tarfiles.py

```diff
@@ -124,19 +124,17 @@ def tar_up(directory: str, excludes: str, file: str = ".") -> str:
     return tarfile
 
 
-def slurp_file(fname: str) -> Tuple[str, bytes]:
+def slurp_file(fname: str) -> Tuple[str, BinaryIO]:
     """pull in a tarfile while computing its hash"""
     h = hashlib.sha256()
-    tfl = []
     with open(fname, "rb") as f:
         tff = f.read(4096)
         h.update(tff)
-        tfl.append(tff)
         while tff:
             tff = f.read(4096)
             h.update(tff)
-            tfl.append(tff)
-    return h.hexdigest(), bytes().join(tfl)
+    tf = open(fname, "rb")
```

Comment:
I'm OK with opening the file twice - I think that's a clean solution.

What really worries me is the readability of opening a file here and closing it in a completely different place.  Is there any reason we can't modify this function to just return the checksum, and then in line 315, pass the filename in as a string to `publisher.publish()`?

Since we're opening the file twice anyway, we could then wrap the `requests.post` call in a file open/close context manager; something like 

```python
with open(tarfile, 'rb') as tarfile_handler:
    res = requests.post(url, auth=TokenAuth(self.token), data=tarfile_handler)
    return res
```

What do you think?

Reply:
I kind of don't like the close-it-elsewhere bit either.. And of course, then this routine becomes *just* a checksummer,
as we're passing the name into it... I'll try that.

Reply:
There, I think that's better.